powerstore/resource_protection_policy.go

```diff
@@ -296,29 +328,103 @@ func (r resourceProtectionPolicy) updatePolicyState(polState *models.ProtectionP
 	polState.Name.Value = polResponse.Name
 	polState.Description.Value = polResponse.Description
 
+	//Update ReplicationRuleIDs value from Response to State
 	var replicationRuleIds []string
 	for _, replicationRule := range polResponse.ReplicationRules {
 		replicationRuleIds = append(replicationRuleIds, replicationRule.ID)
 	}
-	replicationList := []attr.Value{}
+	replicationIDList := []attr.Value{}
 	for _, replicationRuleID := range replicationRuleIds {
-		replicationList = append(replicationList, types.String{Value: string(replicationRuleID)})
+		replicationIDList = append(replicationIDList, types.String{Value: string(replicationRuleID)})
 	}
 	polState.ReplicationRuleIDs = types.List{
 		ElemType: types.StringType,
-		Elems:    replicationList,
+		Elems:    replicationIDList,
 	}
 
+	//Update ReplicationRuleNames value from Plan to State
+	var replicationRuleNames []string
+	for _, replicationRuleName := range polPlan.ReplicationRuleNames.Elems {
+		replicationRuleNames = append(replicationRuleNames, strings.Trim(replicationRuleName.String(), "\""))
+	}
+	replicationNameList := []attr.Value{}
+	for _, replicationRuleName := range replicationRuleNames {
+		replicationNameList = append(replicationNameList, types.String{Value: string(replicationRuleName)})
+	}
+	polState.ReplicationRuleNames = types.List{
+		ElemType: types.StringType,
+		Elems:    replicationNameList,
+	}
+
+	//Update SnapshotRuleIDs value from Response to State
 	var snapshotRuleIds []string
 	for _, snapshotRule := range polResponse.SnapshotRules {
 		snapshotRuleIds = append(snapshotRuleIds, snapshotRule.ID)
 	}
-	snapshotList := []attr.Value{}
+	snapshotIDList := []attr.Value{}
 	for _, snapshotRuleID := range snapshotRuleIds {
-		snapshotList = append(snapshotList, types.String{Value: string(snapshotRuleID)})
+		snapshotIDList = append(snapshotIDList, types.String{Value: string(snapshotRuleID)})
 	}
 	polState.SnapshotRuleIDs = types.Set{
 		ElemType: types.StringType,
-		Elems:    snapshotList,
+		Elems:    snapshotIDList,
+	}
+
+	//Update SnapshotRuleNames value from Plan to State
+	var snapshotRuleNames []string
+	for _, snapshotRuleName := range polPlan.SnapshotRuleNames.Elems {
+		snapshotRuleNames = append(snapshotRuleNames, strings.Trim(snapshotRuleName.String(), "\""))
+	}
+	snapshotNameList := []attr.Value{}
+	for _, snapshotRuleName := range snapshotRuleNames {
+		snapshotNameList = append(snapshotNameList, types.String{Value: string(snapshotRuleName)})
+	}
+	polState.SnapshotRuleNames = types.List{
+		ElemType: types.StringType,
+		Elems:    snapshotNameList,
+	}
+}
+
+func (r resourceProtectionPolicy) fetchByName(plan *models.ProtectionPolicy) (valid bool, err string) {
+	var snapshotRuleIds []string
+	if len(plan.SnapshotRuleIDs.Elems) != 0 && len(plan.SnapshotRuleNames.Elems) != 0 {
+		return false, "Snapshot Rule ID or Snapshot Rule Name"
```

Comment:
can we put the validation from the framework rather than putting it manually?

Reply:
validator support isn't there for v0.10.0. will make these changes while migrating to v1.0